### PR TITLE
Automatically Pull In New Files from RNTester and IntegrationTests

### LINKED ISSUE #5209: Add Rules for Directory Copies to react-native-platform-override
Replaces #4851

Override manifests can currently only describe individual files. When used to sync RNTester, this means we miss any new files added unless we manually add them. To get around this we need to be able to express rules to copy and validate entire directories. Recent re-architecting of the internals of react-native-platform-override should make this much easier.

### PR DESCRIPTION
**Pull request by NickGerleman**
Fixes #5209

This change adds support to override tooling for syncing upstream directories. This lets us automatically get new files added upstream, where before each had to be registered. We get the same benefits of other override tooling, like CI enforcing files are up to date relative to RN version, and protection from tampering.

We don't do anything sophisticated in cases where directories and files overlap right now.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5446)

### CODE REVIEW

File: packages/react-native-platform-override/src/OverrideFactory.ts [hunk 1]
```diff
@@ -92,44 +98,60 @@ export class OverrideFactoryImpl implements OverrideFactory {
...
+      baseDirectory,
       issue,
-      ...(await this.getOverrideBaseInfo(baseFile)),
+      ...(await this.getBaseInfo(baseDirectory)),
     });
   }
 
-  private async checkOverrideFileExists(file: string) {
-    if (path.isAbsolute(file)) {
+  private async checkoverrideFileExists(overridePath: string) {
```
Comment (NickGerleman) [inline packages/react-native-platform-override/src/OverrideFactory.ts:124]: Need to revert naming and consider splitting into file + dir versions.

File: packages/react-native-platform-override/src/Hash.ts [hunk 2]
```diff
@@ -43,9 +44,29 @@ export class Hasher {
...
   }
 
-  return new Hasher().feedContent(content).digest();
+  if (type === 'file') {
+    return hashContent((await repo.readFile(name))!);
+  } else {
+    const hasher = new Hasher();
+    for (const file of (await repo.listFiles([`${name}/**`])).sort()) {
+      const contents = await repo.readFile(file);
+      hasher.feedContent(contents!);
```
Comment (dannyvv) [inline packages/react-native-platform-override/src/Hash.ts:68]: >hasher [](start = 6, length = 6)

I would recommend also hashing the filenames to catch renames that don't affect the sort-order.

Comment (NickGerleman) [reply packages/react-native-platform-override/src/Hash.ts:68]: That's a good point. Will add that.

Comment (NickGerleman) [reply packages/react-native-platform-override/src/Hash.ts:68]: I just realize we don't actually remove old files either. Need to add that as well.

File: packages/react-native-platform-override/src/FileRepository.ts [hunk 3]
```diff
@@ -20,25 +19,29 @@ export interface OverrideFileRepository {
...
-   * doesn't exist.
+   * Check whether a file exists, and if so whether its a file or directory
    */
-  setFileContents(filename: string, content: Buffer): Promise<void>;
+  stat(filename: string): Promise<'file' | 'directory' | 'none'>;
 }
 
 /**
- * Provides access to React Native source files
+ * Provides access to override files
```
Comment (NickGerleman) [inline packages/react-native-platform-override/src/FileRepository.ts:31]: Need to update this comment

**Pull request merged**